### PR TITLE
Implement miscellaneous fixes for partially-defined check

### PR DESCRIPTION
**Pull request by ilinum**
These are the issues that I've found using mypy-primer.

You should be able to review this PR commit-by-commit. Each commit includes the relevant tests:
- Process imports correctly
- Support for function names
- Skip stub files (this change has no tests)
- Handle builtins and implicit module attrs (e.g. `str` and `__doc__`)
- Improved support for lambdas.

### CODE REVIEW

File: mypy/partially_defined.py [hunk 1]
```diff
@@ -396,3 +418,19 @@ def visit_with_stmt(self, o: WithStmt) -> None:
             expr.accept(self)
             self.process_lvalue(idx)
         o.body.accept(self)
+
+    def visit_import(self, o: Import) -> None:
+        for mod, alias in o.ids:
```
Comment (ilevkivskyi) [inline mypy/partially_defined.py:423]: I think you need to split `id`s with `.` (not 100% sure)

File: test-data/unit/check-partially-defined.test [hunk 2]
```diff
@@ -432,3 +468,53 @@ def f5() -> None:
         y = z  # E: Name "z" is used before definition
         x = z  # E: Name "z" is used before definition
     z: int = 2
+
+[case testUseBeforeDefImports]
+# flags: --enable-error-code use-before-def
+from foo import x  # type: ignore
+from foo import abc as y  # type: ignore
+import z  # type: ignore
+import foo as f  # type: ignore
```
Comment (ilevkivskyi) [inline test-data/unit/check-partially-defined.test:477]: Could you please add a test that `import foo.bar` makes `foo` defined?

**Review by ilevkivskyi — APPROVED** [review]
LG, but there are couple questions about dotted imports.

File: mypy/partially_defined.py (hunk 1 above)
Comment (ilinum) [reply mypy/partially_defined.py:423]: Good catch on both of these! Done.

**Comment by ilinum** [comment]
> LG, but there are couple questions about dotted imports.

Good catch on these! Thanks!

File: mypy/partially_defined.py [hunk 3]
```diff
@@ -396,3 +418,20 @@ def visit_with_stmt(self, o: WithStmt) -> None:
...
+            names = mod.split(".")
+            if alias is not None:
+                names[-1] = alias
+            for name in names:
+                self.tracker.record_definition(name)
```
Comment (ilevkivskyi) [inline mypy/partially_defined.py:428]: Actually I don't think this logic is correct, after `import x.y` only `x` becomes defined (try `import os.path; path` at runtime). While if you do `import x.y as z`, then only `z` becomes defined.

Comment (ilinum) [reply mypy/partially_defined.py]: Yep, you're right! I had no idea. Fixed

**Comment by ilinum** [comment]
@ilevkivskyi I'm going to merge this because I'm pretty sure it's correct now but please take a look at the imports implementation if you get the chance :) I'll send a follow-up PR if something is actually broken.

**Pull request merged**